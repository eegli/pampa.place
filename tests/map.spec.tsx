import { initialize, Map } from '@googlemaps/jest-mocks';
import React from 'react';
import GoogleMap, { MapMode } from '../src/components/google/google.map';
import { testMap } from './payloads/map-data';
import { render } from './test-utils';

beforeEach(() => {
  initialize();
});

jest.spyOn(React, 'useRef').mockReturnValue({
  current: {
    style: {
      display: '',
    },
  },
});

// @ts-expect-error
Map.prototype.data = {
  addGeoJson: jest.fn().mockImplementation(() => []),
  setStyle: jest.fn(),
};

describe('Google Map', () => {
  it('renders', () => {
    const { container } = render(
      <GoogleMap mapData={testMap} mode={MapMode.PREVIEW} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
