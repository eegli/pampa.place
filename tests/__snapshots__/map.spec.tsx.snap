// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Map renders 1`] = `
<div
  id="__GMAP__CONTAINER__"
  style="width: 100%; height: 100%; display: none;"
/>
`;
